### PR TITLE
Adds delegation feature

### PR DESCRIPTION
**Pull request by betsyecastro**
And adds delegates and delegators relations to the User.php model.

### CODE REVIEW

File: app/User.php [hunk 1]
```diff
@@ -326,4 +327,116 @@ public function bookmarked($model)
...
+
+    /**
+     * Current user delegators.
+     * 
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function currentDelegators()
+    {
+        return $this->delegators()
+            ->where('user_delegations.starting', '<=', DB::raw('now()'))
```
Comment (wunc) [inline app/User.php:405]: Maybe change to `->whereDate('user_delegations.starting', '<=', now())` and similar on line 407.

File: app/User.php [hunk 2]
```diff
@@ -326,4 +327,116 @@ public function bookmarked($model)
...
+    public function currentDelegatedRoles()
+    {
+        return Role::query()
+            ->wheredoesntHave('users', function ($q) {
+                $q->where('id', '=', $this->id);
+            })
+            ->whereHas('users.delegates', function ($q) {
+                $q->where(function ($q) {
+                    $q->where('delegate_user_id', $this->id);
+                    $q->where('user_delegations.until', '>', DB::raw('now()'));
```
Comment (wunc) [inline app/User.php:436]: This and the next orWhere should probably be grouped.

**Review by wunc — CHANGES_REQUESTED** [review]

File: app/User.php (hunk 1 above)
Comment (betsyecastro) [reply app/User.php]: @wunc This correction was applied in next commit.

File: app/User.php (hunk 2 above)
Comment (betsyecastro) [reply app/User.php]: @wunc This correction was applied in the next commit.

**Review by wunc — APPROVED** [review]

**Pull request merged**